### DISCUSSION TITLE
When to import through Geoserver and when import with "Upload dataset" through the Geonode interface

### DISCUSSION THREAD

**Discussion by lcoandrade** [GeoNode/geonode#11482] [Q&A]
I would like to know the a good approach to import data. When import through Geoserver and when import with "Upload dataset" through the Geonode interface.

Context:
I need to deploy a Geonode at my University, something like https://my_uni/geonode. The university will provide me a virtual machine to make this happen and I will use a docker installation for this Geonode.

Problem:
Here comes the question. Considering the need to import data generated by all projects and thesis, the need to enter in the VM and copy data using `docker cp /path/to/my/data/ geoserver4geonode:/geoserver_data/data` can be cumbersome. But, this approach would made the use of OGC services available.

On the other hand, import with "Upload dataset" through the Geonode interface would not give this option. Am I right?

So, if the Department I'm working with assigns user accounts to some managers, it would be better to use "Upload dataset" directly instead of doing the import through Geoserver. But what are the advantages and disadvantages to each of these options?

Is it possible to set OGC Services for data imported with "Upload dataset" through the Geonode interface?

**Comment by gannebamm** [comment]
Hi Luiz,
I work at a federal research institute, and we also have loads of projects providing data for our GeoNode instances. We often explain to the working groups how to use the fronted UI to upload their datasets; many of them can use the UI system to upload the data. That is the preferred way. Sometimes, this will fail due to huge filesize and special needs or file formats (netCDF, e.g.). Then, we take care of the uploading process and manually sent those datasets. If the datasets are vector/database tables, we also provide a database landing zone at which the researchers could push their data. My team then takes a look at the tables in the landing zone and copies those to the geonode_data database. To register the manually uploaded/ingested datasets in GeoNode, you can use the async tasks in the backend (Django Admin) interface. Furthermore, there is a command-line tool developed by a partner research institution called geonode_ctl (https://github.com/GeoNodeUserGroup-DE/geonodectl/tree/main). That may help in bulk uploads, too.

Was this information helpful?

**Reply by lcoandrade** [reply]
Hi,  @gannebamm!

Your answer was really helpful. I think it is a good approach. Is this process of yours documented? How the professionals get to know how to proceed? Do you have a published flowchart? I'm thinking to publish something like this in the geonode itself.
